frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -2521,6 +2526,78 @@ private void serveLicenseToSourceFileMapping(ResourceRequest request, ResourceRe
+        final String projectId = request.getParameter(PROJECT_ID);
+        final JSONObject jsonResult = createJSONObject();
+
+        try {
+            Project project = client.getProjectById(projectId, user);
+            Set<String> releaseIds = CommonUtils.getNullToEmptyKeyset(project.getReleaseIdToUsage());
+            StringBuilder oneCLI = new StringBuilder();
+            StringBuilder multipleCLI = new StringBuilder();
+            StringBuilder noCLI = new StringBuilder();
+
+            for (String releaseId : releaseIds) {
+                Release release = componentClient.getReleaseById(releaseId, user);
+                List<Attachment> filteredAttachments = SW360Utils.getApprovedClxAttachmentForRelease(release);
+                Set<String> mainLicenses = release.getMainLicenseIds() == null ? new HashSet<String>() : release.getMainLicenseIds();
+                Set<String> otherLicenses = release.getOtherLicenseIds() == null ? new HashSet<String>() : release.getOtherLicenseIds();
+
+                if (filteredAttachments.size() == 1) {
+                    final Attachment attachment = filteredAttachments.get(0);
+                    final String attachmentName = attachment.getFilename();
+                    oneCLI.append(CommonUtils.nullToEmptyString(printName(release))).append(",");
+                    List<LicenseInfoParsingResult> licenseInfoResult = licenseInfoClient.getLicenseInfoForAttachment(release,
+                            attachment.getAttachmentContentId(), true, user);
+                    List<LicenseNameWithText> licenseWithTexts = licenseInfoResult.stream()
+                            .flatMap(result -> result.getLicenseInfo().getLicenseNamesWithTexts().stream())
+                            .filter(license -> !license.getLicenseName().equalsIgnoreCase(SW360Constants.LICENSE_NAME_UNKNOWN)
+                                    && !license.getLicenseName().equalsIgnoreCase(SW360Constants.NA)
+                                    && !license.getLicenseName().equalsIgnoreCase(SW360Constants.NO_ASSERTION)) // exclude unknown, n/a and noassertion
+                            .collect(Collectors.toList());
+                    if (attachmentName.endsWith(".rdf")) {
+                        mainLicenses.addAll(licenseInfoResult.stream()
+                                .flatMap(singleResult -> singleResult.getLicenseInfo().getConcludedLicenseIds().stream())
+                                .collect(Collectors.toSet()));
+                        otherLicenses.addAll(licenseWithTexts.stream().map(LicenseNameWithText::getLicenseName).collect(Collectors.toSet()));
+                        otherLicenses.removeAll(mainLicenses);
+                    } else if (attachmentName.endsWith(".xml")) {
+                        mainLicenses.addAll(licenseWithTexts.stream()
+                                .filter(license -> license.getType().equals(LICENSE_TYPE_GLOBAL))
```

Comment:
May be define a predicate for this `license -> license.getType().equals(LICENSE_TYPE_GLOBAL)` and negate it in line number 2573

Reply:
added the suggested change.